core/chains/evm/logpoller/log_poller.go

```diff
@@ -260,6 +369,16 @@ func (lp *logPoller) Start(parentCtx context.Context) error {
 		ctx, cancel := context.WithCancel(parentCtx)
 		lp.ctx = ctx
 		lp.cancel = cancel
+		filters, err := lp.orm.LoadFilters(pg.WithParentCtx(lp.ctx))
+		lp.filterDirty = true
+		if err != nil {
+			// Most likely, we will get them anyway once RegisterFilter() is called, but
```

Comment:
the comment confuses me.

why wouldn't start fail if loading the filters fails? is as, can it lead to an inconsistent state if there an error?

Reply:
It could fail due to a timeout if there is a temporary network outage, or the db or the node is too overloaded with work and slow at processing requests.  Or maybe even some subtle error in the SQL statement or decoding/encoding of params, perhaps that depends on the version of postgres.  Later, when the job starts it will try to add the filter, but the job itself constructs the filter so that doesn't depend on the db.  Some of those could be indications of more widespread problem that would cause the job to be unable to start.  Perhaps "most likely" is overly optimistic, but either way it seems best to assume it will start okay instead of refuse to start logpoller which would in itself prevent the job from being able to run properly.

You could consider it an inconsistent state in that before the job starts, the filter would be in db but not in memory, but that would be fixed as soon as the job starts.  In fact, that would be no different from current behavior... we start logpoller with no filters in memory, then let the individual job services add them when they start.

Reply:
hmm I'd treat this the same as SelectLatestBlock, its just DB state we want to load before starting to poll. We can keep retrying forever in the log poll tick loop and only start progressing from last saved block once we've successfully loaded the filters. Then we can cleanly recover from transient DB issues and never miss any logs

Reply:
Kind of a general pattern for synchronizing external state I've seen a bunch. 1) On boot try forever to load initial state and a progress identifier (i.e. block pointer) from DB 2) While up atomically update state and block pointer in a DB tx. Doesn't matter if the node or DB crashes/flaps up and down etc. you never miss anything or have any partial updates.